plugins/wporg-5ftf/includes/pledge-log.php

```diff
@@ -0,0 +1,322 @@
+<?php
+namespace WordPressDotOrg\FiveForTheFuture\PledgeLog;
+
+use WordPressDotOrg\FiveForTheFuture;
+use WordPressDotOrg\FiveForTheFuture\{ Contributor, Pledge, PledgeForm, PledgeMeta };
+use WP_Post;
+
+defined( 'WPINC' ) || die();
+
+const LOG_META_KEY = FiveForTheFuture\PREFIX . '_activity-log';
+
+// Log display.
+add_action( 'admin_init', __NAMESPACE__ . '\add_log_meta_box' );
+
+// Log capture.
+add_action( 'save_post_' . Pledge\CPT_ID, __NAMESPACE__ . '\capture_save_post', 99, 3 );
+add_action( 'updated_postmeta', __NAMESPACE__ . '\capture_updated_postmeta', 99, 4 );
+add_action( 'added_post_meta', __NAMESPACE__ . '\capture_added_post_meta', 99, 4 );
+add_action( 'transition_post_status', __NAMESPACE__ . '\capture_transition_post_status', 99, 3 );
+add_action( FiveForTheFuture\PREFIX . '_add_pledge_contributors', __NAMESPACE__ . '\capture_add_pledge_contributors', 99, 3 );
+add_action( FiveForTheFuture\PREFIX . '_remove_contributor', __NAMESPACE__ . '\capture_remove_contributor', 99, 3 );
+
+/**
+ * Adds a meta box for the log on the custom post type.
+ *
+ * @return void
+ */
+function add_log_meta_box() {
+	add_meta_box(
+		'activity-log',
+		__( 'Log', 'wordpressorg' ),
+		__NAMESPACE__ . '\render_log_meta_box',
+		Pledge\CPT_ID,
+		'advanced',
+		'low'
+	);
+}
+
+/**
+ * Output the log in the meta box.
+ *
+ * @param WP_Post $pledge
+ *
+ * @return void
+ */
+function render_log_meta_box( $pledge ) {
+	$log = get_pledge_log( $pledge->ID );
+
+	require FiveForTheFuture\get_views_path() . 'log.php';
+}
+
+/**
+ * Defaults for a log entry.
+ *
+ * @return array {
+ *     @type int    $timestamp Time of the event.
+ *     @type string $type      The type of event. A snake_case or kebab-case string.
+ *     @type string $message   Description of the event.
+ *     @type array  $data      Details and data related to the event.
+ *     @type int    $user_id   The ID of the logged in user who triggered the event, if applicable.
+ * }
+ */
+function get_log_entry_template() {
+	return array(
+		'timestamp' => time(),
```

Comment:
`MAYBE` - It might be useful to use `microtime()` here, since there could be some events (like inserting multiple contributors) that will happen during the same request.

Reply:
If there are multiple entries with the same timestamp in seconds, I think that's ok. If we need to investigate a race condition, we can switch or add another key to the array with microtime.